### PR TITLE
Add integration tests for important commands

### PR DESCRIPTION
**Pull request by pjcdawkins**
Tested on the GitLab side

Rationale:
- Go seems to be fine at running these kinds of tests, and it removes the need to worry about Python and Selenium, which have not been working (on GitLab), and which represent unnecessary barriers to testing on other platforms.
- Go allows easy mocking of API servers to test particular commands.

Awkwardnesses were:
- I set `cmd.Dir = os.TempDir()` when running a command: so that the legacy CLI doesn't do unpredictable things with the "current project".
- I added the `chi` router package as a dependency for the test mocks. That probably does not affect the production build (on this branch, it's about 1KB smaller for me).

Follow up steps could be:

- Build or download PHP and the Legacy CLI on GitHub Actions, so that the integration tests can be run there.
- Test on multiple operating systems, such as Linux, Windows and MacOS.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -84,10 +84,17 @@ release: goreleaser clean-phar internal/legacy/archives/platform.phar php ## Rel
 .PHONY: test
 test: ## Run unit tests
 	go clean -testcache
-	go test -v -race -mod=readonly -cover ./...
+	go test -v -race -short -cover ./...
+
+platform: internal/legacy/archives/platform.phar php
+	go build ./cmd/platform
```
[the remark this answers is not in the document]
Comment (pjcdawkins) [reply Makefile:90]: The only issue is in then finding the executable afterwards. It's not clear how GoReleaser constructs the file path - e.g. where does the `v1` come from in `./dist/platform_linux_amd64_v1/platform` ?

File: tests/tests.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+	}
+	versionCmd := exec.Command(candidate, "--version")
+	versionCmd.Env = testEnv()
+	output, err := versionCmd.Output()
+	require.NoError(t, err, "running '--version' must succeed under the CLI at: %s", candidate)
+	require.Contains(t, string(output), "Platform Test CLI ")
+	t.Logf("Validated CLI command %s", candidate)
+	_validatedCommand = candidate
+	return _validatedCommand
+}
```
[the remark this answers is not in the document]
Comment (pjcdawkins) [reply tests/tests.go:55]: Just that we need to test a particular build, not the globally installed `platform`

[the remark this answers is not in the document]
Comment (pjcdawkins) [reply tests/tests.go:55]: The alternative I wondered about is just calling `commands.Execute()` - which could result in more accurate coverage info (if you grep out the `mockapi` and `tests` directories from coverage).

File: Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (akalipetis) [reply Makefile:90]: It comes from here: https://goreleaser.com/customization/builds/#why-is-there-a-_v1-suffix-on-amd64-builds

You can expect it to be like this for the foreseeable future.

File: tests/tests.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (akalipetis) [reply tests/tests.go:55]: That's true, but I find it a bit more difficult to follow - why don't we just shell out to `./dist/platform_linux_amd64_v1/platform` using the provided arguments?

File: Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (pjcdawkins) [reply Makefile:90]: There's a lot of hardcoding which I wouldn't want to reproduce here :/  https://github.com/goreleaser/goreleaser/blob/6a84e401158d4d144bf9c48f83fb4f699f84cab2/internal/builders/golang/build.go#L59-L92

[the remark this answers is not in the document]
Comment (pjcdawkins) [reply Makefile:90]: I could get around that by globbing for `dist/platform_*/platform` and running whichever is found first?

And.. unfortunately for the Makefile `single` is phony / runs every time, and takes a little while.

[the remark this answers is not in the document]
Comment (pjcdawkins) [reply Makefile:90]: For the first part see https://github.com/platformsh/cli/pull/214/files

File: tests/tests.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (pjcdawkins) [reply tests/tests.go:55]: That is what we're doing for GitLab where TEST_CLI_PATH is provided explicitly. The rest of the problem is about knowing the exact path - see the other comment thread.

**Review by akalipetis — APPROVED** [review]
LGTM

**Pull request merged**